create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -424,12 +427,23 @@ private Contributor generateContributorFromAuthorTp(AuthorTp author, AuthorGroup
 
     private Optional<Map<String, String>> getOrganizationLabels(AuthorGroupTp authorGroup) {
         var organizationNameOptional = getOrganizationNameFromAuthorGroup(authorGroup);
-        return organizationNameOptional.map(organizationName -> Map.of(getLanguageIso6391Code(),
+        return organizationNameOptional.map(organizationName -> Map.of(getLanguageIso6391Code(organizationName),
                                                                        organizationName));
     }
 
-    private String getLanguageIso6391Code() {
-        return "en";
+    private String getLanguageIso6391Code(String textToBeGuessedLanguageCodeFrom) {
+        var detector = new OptimaizeLangDetector().loadModels();
+        var result = detector.detect(textToBeGuessedLanguageCodeFrom);
+        return result.isReasonablyCertain()
+                   ? getIso6391LanguageCodeForSupportedNvaLanguage(result.getLanguage())
+                   : ENGLISH.getIso6391Code();
```

Comment:
Occurs to me that we may have similar issues with "main language" for publications, though there, we have a different solution…